src/main/webapp/js/finscholar/appstate/appstate.js

```diff
@@ -108,13 +109,44 @@ class AppState {
       this.currentView_.registerListener(this.listViewUpdate_.bind(this));
     }
     this.currentView_.renderView();
+    this.currentIndex_ = index;
     this.refreshNavbar_();
   }
 
+  /** 
+   * Updates the classlist of navbar buttons to indicate 
+   * to users which tab has been selected. 
+   * @private
+   */
+  selectTab_() {
+    switch (this.currentIndex_) {
+      case 0:
+        googDom.getElement('home').classList.add('active');
+        googDom.getElement('collegesList').classList.remove('active');
+        googDom.getElement('scholarshipsList').classList.remove('active');
```

Comment:
A couple things.
1) This pattern isn't maintainable. Where do the indices come from? If you add a button or change the order you have to fix all this code.
2) You can condense this entire switch block down to 2 lines by selecting all buttons by class name and removing the class. Then add the class to the specific button.

Reply:
The index comes from the way the navbar is managed. It's honestly kinda wonky, and the navbar would have to be redesigned to use something different than a numerical index (it also gets refreshed each time content changes). For now I'll make a map that maps the indices to button id's and I'll uses that to add the 'active' class to the classlist of the selected tab.